server/src/app/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,220 @@
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
+    });
+    it('should throw UnauthorizedException()', async () => {
+      const mockService: AuthService = mock(AuthService);
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(true);
+
+      when(mockUserService.create(address, name))
+        .thenReturn(new Promise((resolve) => {
+          resolve(undefined);
+        }));
+
+      when(mockService.createToken(address))
+        .thenReturn(undefined);
+
+      when(mockJwtService.sign(address)).thenReturn(undefined);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
+    });
+  });
+
+  describe('#signIn', () => {
+    it('should return a user', async () => {
+
+      const mockService: AuthService = mock(AuthService);
+
+      const user: User = new User(address, name);
+      const jwt: Token = {
+        expiresIn: 3600,
+        accessToken: 'token',
+      };
+
+      const signInResponseDto: SignInResponseDto
+        = new SignInResponseDto(accessToken, address, name);
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(true);
+
+      when(mockUserService.getByAddress(address))
+        .thenReturn(new Promise((resolve) => {
+          resolve(user);
+        }));
+
+      when(mockService.createToken(address))
+        .thenReturn(jwt);
+
+      when(mockJwtService.sign(address)).thenReturn(accessToken);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signIn(address, message, signature)).resolves.toEqual(signInResponseDto);
+    });
+    it('should throw UnauthorizedException()', async () => {
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(false);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
+    });
+    it('should throw "user not found"', async () => {
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(true);
+
+      when(mockUserService.getByAddress(address))
+        .thenReturn(new Promise((resolve) => {
+          resolve(undefined);
+        }));
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
+    });
+    it('should throw UnauthorizedException', async () => {
```

Comment:
u don't have to test this cause u already tested the case of validationService to return false